crates/pop-cli/src/common/mod.rs

```diff
@@ -16,3 +16,219 @@ pub mod prompt;
 /// Contains runtime utilities.
 pub mod runtime;
 pub mod wallet;
+
+use std::fmt::{Display, Formatter, Result};
+use strum::VariantArray;
+
+#[derive(Debug, PartialEq)]
+pub enum Telemetry {
+	Null,
+	Build(Project),
+	Test { project: Project, feature: Feature },
+	Install(Os),
+	Up(Project),
+	New(Template),
+}
+
+/// Represents the type of project being operated on.
+#[derive(Debug, PartialEq, Clone, VariantArray)]
+pub enum Project {
+	Network,
+	/// A blockchain project (parachain, etc.).
+	Chain,
+	/// A smart contract project.
+	Contract,
+	/// Unknown project type.
+	Unknown,
+}
+
+#[derive(Debug, PartialEq)]
+pub enum Template {
+	Chain(pop_parachains::Parachain),
+	Contract(pop_contracts::Contract),
+	Pallet,
+}
+
+#[derive(Debug, PartialEq, Clone, VariantArray)]
+pub enum Feature {
```

Comment:
maybe a different name for this -- variants are all around testing? Something like: "TestFeatures", or something like that.